horiz_interp/horiz_interp_type.F90

```diff
@@ -189,35 +189,35 @@ subroutine horiz_interp_type_eq(horiz_interp_out, horiz_interp_in)
       call mpp_error(FATAL,'horiz_interp_type_eq: horiz_interp_type variable on right hand side is unassigned')
     endif
 
-    horiz_interp_out%faci            => horiz_interp_in%faci
-    horiz_interp_out%facj            => horiz_interp_in%facj
-    horiz_interp_out%ilon            => horiz_interp_in%ilon
-    horiz_interp_out%jlat            => horiz_interp_in%jlat
-    horiz_interp_out%area_src        => horiz_interp_in%area_src
-    horiz_interp_out%area_dst        => horiz_interp_in%area_dst
-    horiz_interp_out%wti             => horiz_interp_in%wti
-    horiz_interp_out%wtj             => horiz_interp_in%wtj
-    horiz_interp_out%i_lon           => horiz_interp_in%i_lon
-    horiz_interp_out%j_lat           => horiz_interp_in%j_lat
-    horiz_interp_out%src_dist        => horiz_interp_in%src_dist
+    horiz_interp_out%faci            =  horiz_interp_in%faci
+    horiz_interp_out%facj            =  horiz_interp_in%facj
+    horiz_interp_out%ilon            =  horiz_interp_in%ilon
+    horiz_interp_out%jlat            =  horiz_interp_in%jlat
+    horiz_interp_out%area_src        =  horiz_interp_in%area_src
+    horiz_interp_out%area_dst        =  horiz_interp_in%area_dst
+    horiz_interp_out%wti             =  horiz_interp_in%wti
+    horiz_interp_out%wtj             =  horiz_interp_in%wtj
+    horiz_interp_out%i_lon           =  horiz_interp_in%i_lon
+    horiz_interp_out%j_lat           =  horiz_interp_in%j_lat
+    horiz_interp_out%src_dist        =  horiz_interp_in%src_dist
     horiz_interp_out%found_neighbors => horiz_interp_in%found_neighbors
```

Comment:
Did you miss this pointer?

Reply:
I couln't find any use of this pointer in FMS.  Specifically if you grep the repository, you only get the following lines:
https://github.com/NOAA-GFDL/FMS/blob/203c8bf464ff26fe0fe39b1451caedd026bbce55/horiz_interp/horiz_interp_type.F90#L72
https://github.com/NOAA-GFDL/FMS/blob/203c8bf464ff26fe0fe39b1451caedd026bbce55/horiz_interp/horiz_interp_type.F90#L203

So, since it wasn't a pointer being allocated I did not touch that variable at all.

Reply:
Does my reasoning make sense?

Reply:
It seems odd because the routine is creating a copy except for this one variable.

Reply:
I'd bet its just unused in general, you don't get warnings for unused fields in derived types so it could have never been used and no one noticed. I think we can probably remove it unless we can find some code that actually uses it.

Reply:
If we remove this field assignment, it will violate principle of object-to-object copy but if it is never used, we can remove. We can add something like this:

if (associated(in%var)) 
    if (associated(out%var)) deallocate(out%var)
    allocate(out%var(size(in%var))
    out%var = in%var
else
    out%var => NULL()
end if

Reply:
@thomas-robinson @ganganoaa @rem1776 i have not done anything to found neighbors yet.  the options as I see it are:
- removing it seems like something that would require a lot of testing since its possible some model somewhere is using it.  
- changing it from pointer to allocatable change this pointer assignment to equals
- leave it as is
I don't feel like I have enough familiarity with interpolator code to make this decision.  How should I proceed?   Who can I consult on what to do?

Reply:
do this:
```fortran
if (allocated(horiz_interp_in%found_neighbors))horiz_interp_out%found_neighbors = horiz_interp_in%found_neighbors
```
and make `found_neighbors` allocatable instead of a pointer.  This is a `=` routine, so it should be making a copy, not providing pointers.  

This is a modernization PR, not a clean up.  No need to remove it here.  We should note it for potential clean up in a future release.